src/index.js

```diff
@@ -154,19 +155,34 @@ function renderPdf(tag) {
   }
 
   const result = {};
+  const isTopLevel = true;
 
   children.forEach(child => {
     const resolvedChild = resolve(child, context);
     result[resolvedChild.elementName] = resolveChildren(
       resolvedChild,
       context,
-      true,
+      isTopLevel,
     );
   });
 
+  const documentFnAttrs = ['footer', 'header'].reduce((memo, key) => {
+    /* eslint-disable-next-line no-param-reassign */
+    memo[key] = result[key];
+
+    if (has(attributes, key) && typeof attributes[key] === 'function') {
+      /* eslint-disable-next-line no-param-reassign */
```

Comment:
To avoid disabling eslint here, I suggest we spread the function return value:

```javascript
return {
  ...memo,
  key: (...args) => 
    resolveChildren(attributes[key](...args), context, isTopLevel),
};
```

Reply:
We could do that but it's much less performant. Probably doesn't matter too much since this is a small loop

Reply:
Since this object can only ever have up to 2 properties, and they're only ever set to function references, I think performance is probably not worth worrying about. If we were dealing with an object that could contain the whole document, I'd definitely agree that we want to avoid allocating a new object every iteration, since the document can be arbitrarily large.